sympy/solvers/solvers.py

```diff
@@ -256,6 +256,11 @@ def checksol(f, symbol, sol=None, **flags):
             rv = None  # don't return, wait to see if there's a False
         return rv
 
+    f = _sympify(f)
+
+    if not f.free_symbols:
+        return f.is_zero
```

Comment:
Since the routine will try different strategies, shouldn't this be allowed to continue? It would only be safe to do this `if f.is_Number:`